### ISSUE TITLE
Unable to set node name with ServerRunner

### ISSUE THREAD

**Issue by mwasserschaff** [chefspec/chefspec#776]
Chefspec does not allow to modify the node name. It is always "chefspec" though #726 indicates that it should be possible.

```
require 'chefspec'

describe 'custom client name' do
  context 'serverrunner' do
    let(:chef_run) do
      runner = ChefSpec::ServerRunner.new do |node, server|
        Chef::Config[:node_name] = "mynode"
        Chef::Config[:client_name] = "mynode"
      end
      runner.converge()
    end

    it 'has node name set' do
      expect(chef_run.node.name).to eq("mynode")
    end
  end
end
```

fails with:
```
 1) custom client name serverrunner has node name set                                                                                                                                                   
     Failure/Error: expect(chef_run.node.name).to eq("mynode")                                                                                                                                           
                                                                                                                                                                                                         
       expected: "mynode"                                                                                                                                                                                
            got: "chefspec"                                                                                                                                                                              
```

**Comment by joerg** [comment]
I was able to set the node name like this:

```
    let(:chef_run) do
      runner = ChefSpec::ServerRunner.new(@server_runner_opts_rhel6) do |node, _server|
        node.name 'my.name'
        node.chef_environment '_default'
      end
      runner.converge(described_recipe)
```

Setting the environment was necessary in my case because otherwise it would be empty and some cookbook code would fail.

@tas50 Maybe it would be a good idea to include a full example that sets various ServerRunner specifics like environments, data_bags, node name, file cache path etc. Those are all documented somewhere, but I found most of it in bug reports and blog posts.
